### DISCUSSION TITLE
Understanding OS CPU Usage

### DISCUSSION THREAD

**Discussion by arcarmona** [emqx/emqx#13249] [Q&A] [answered]
I have a emqx cluster running behind a HAProxy acting as load balancer with three nodes each with 3.5 cpu (previously using two and experiencing the same issue) where node 0  seeems to be always carrying a noticeably higher cpu load as shown by the OS CPU Load metric in the dashboard, however when checking the cpu usage of beam.smp within each node they all seem to have similar values of 250%-270% so my question is:
1. How exactly is the load shown in the dashboard calculated and how should it be interpreted?
2. What exactly could be causing this imbalance in load towards the first one or what steps could be taken to debug it?

**Comment by zmstone** [comment, marked as answer]
Hi @arcarmona 
The numbers from Erlang/OTP's builtin CPU supervisor, for which you can read more here: https://www.erlang.org/docs/26/man/cpu_sup

I did not study it in depth, but it appears `cpu_sup` simply reads the numbers from `/proc/loadavg` https://github.com/simplegeo/erlang/blob/15eda8de27ba73d176c7eeb3a70a64167f50e2c4/lib/os_mon/src/cpu_sup.erl#L227-L238

Imbalance usually caused by unevenly distributed client connections.
e.g. when you perform a rolling restart of the cluster nodes, 
after a round of restart, the last node tend to be least loaded, and the first restarted tend to be the most loaded.

consider using EMQX enterprise which has a cluster rebalance feature: https://docs.emqx.com/en/enterprise/latest/deploy/cluster/rebalancing.html#node-evacuation-and-cluster-load-rebalancing
